standard/expressions.md

```diff
@@ -4339,6 +4374,54 @@ A *throw expression* shall only occur in the following syntactic contexts:
 - As the second operand of a null coalescing operator (`??`).
 - As the body of an expression-bodied lambda or member.
 
+## §declaration-expressions-new-clause Declaration expressions
```

Comment:
We may want to make a note that this is legal:

```
public class C {
    public void M(out int i) {
    }
    
    public void M2()
    {
        String _ = "bad";
        M(out int _);
}
```

The point being that the first `_` is a variable named `_`, and the declaration expression introduces a variable, but does not introduce a name.

Reply:
Example added in upcoming commit